diff --git a/.bash_aliases b/.bash_aliases
new file mode 100644
index 000000000..4349caf18
--- /dev/null
+++ b/.bash_aliases
@@ -0,0 +1 @@
+alias regenerate_baserow_patch='git diff baserow-1.33.4..baserow-vocabai-patch-1.33.4 | sed "s:/:/:g" > ~/code/python/baserow-vocabai-plugin/plugins/baserow_vocabai_plugin/baserow-patches/baserow.patch'
diff --git a/backend/docker/docker-entrypoint.sh b/backend/docker/docker-entrypoint.sh
index aa6e1cb5c..bfa80a22a 100755
--- a/backend/docker/docker-entrypoint.sh
+++ b/backend/docker/docker-entrypoint.sh
@@ -390,6 +390,11 @@ case "$1" in
       echo "Running celery export worker healthcheck..."
       exec celery -A baserow inspect ping -d "export-worker@$HOSTNAME" -t 10 "${@:2}"
     ;;
+    celery-cloudlanguagetoolsworker)
+      BASEROW_CELERY_CLT_WORKER_NUM=${BASEROW_CELERY_CLT_WORKER_NUM:-2}
+      echo "Running celery cloudlanguagetools worker with ${BASEROW_CELERY_CLT_WORKER_NUM} workers..."
+      start_celery_worker --concurrency ${BASEROW_CELERY_CLT_WORKER_NUM} -Q cloudlanguagetools -n cloudlanguagetools-worker@%h "${@:2}"
+    ;;    
     celery-beat)
       # Delay the beat startup as there seems to be bug where the other celery workers
       # starting up interfere with or break the lock obtained by it. Without this the
diff --git a/backend/pytest.ini b/backend/pytest.ini
index 082f2c2cd..a20409be3 100644
--- a/backend/pytest.ini
+++ b/backend/pytest.ini
@@ -1,7 +1,7 @@
 [pytest]
 DJANGO_SETTINGS_MODULE = baserow.config.settings.test
 python_files = test_*.py
-addopts = --disable-warnings
+addopts = --disable-warnings --reuse-db
 env =
     SECRET_KEY = test
     BASEROW_JWT_SIGNING_KEY = test
diff --git a/backend/src/baserow/config/settings/base.py b/backend/src/baserow/config/settings/base.py
index dc121dd86..05b009b05 100644
--- a/backend/src/baserow/config/settings/base.py
+++ b/backend/src/baserow/config/settings/base.py
@@ -181,6 +181,8 @@ CELERY_TASK_ROUTES = {
     "baserow.core.usage.tasks": {"queue": BASEROW_GROUP_STORAGE_USAGE_QUEUE},
     "baserow.contrib.database.table.tasks.run_row_count_job": {"queue": "export"},
     "baserow.core.jobs.tasks.clean_up_jobs": {"queue": "export"},
+    # cloudlanguagetools jobs go on a separate queue
+    "*cloudlanguagetools*": {"queue": "cloudlanguagetools"},
 }
 CELERY_SOFT_TIME_LIMIT = 60 * 5  # 5 minutes
 CELERY_TIME_LIMIT = CELERY_SOFT_TIME_LIMIT + 60  # 60 seconds
diff --git a/backend/src/baserow/core/models.py b/backend/src/baserow/core/models.py
index bf7ec4373..d97857af0 100755
--- a/backend/src/baserow/core/models.py
+++ b/backend/src/baserow/core/models.py
@@ -216,7 +216,8 @@ class UserProfile(models.Model):
     # TODO Remove null=True in a future release.
     email_verified = models.BooleanField(null=True, default=False)
     # TODO Remove null=True in a future release.
-    completed_onboarding = models.BooleanField(null=True, default=False)
+    # vocab-words-patch: we don't want to do onboarding for new users
+    completed_onboarding = models.BooleanField(null=True, default=True)
     completed_guided_tours = ArrayField(
         base_field=models.TextField(),
         default=list,
diff --git a/backend/src/baserow/core/posthog.py b/backend/src/baserow/core/posthog.py
index 42f7ea5c8..1e95c3dea 100644
--- a/backend/src/baserow/core/posthog.py
+++ b/backend/src/baserow/core/posthog.py
@@ -30,7 +30,7 @@ def capture_event(distinct_id: str, event: str, properties: dict):
     try:
         posthog.capture(
             distinct_id=distinct_id,
-            event=event,
+            event=f'baserow:{event}',
             properties=properties,
         )
     except Exception as e:
@@ -77,7 +77,7 @@ def capture_user_event(
     if workspace is not None:
         properties["workspace_id"] = workspace.id
 
-    capture_event(user_id, event, properties)
+    capture_event(user.email, event, properties)
 
 
 @receiver(action_done)
diff --git a/supervisor/default_baserow_env.sh b/supervisor/default_baserow_env.sh
index 5e4453f10..2a88d905d 100755
--- a/supervisor/default_baserow_env.sh
+++ b/supervisor/default_baserow_env.sh
@@ -71,6 +71,7 @@ export DJANGO_SETTINGS_MODULE=${DJANGO_SETTINGS_MODULE:-$DEFAULT_DJANGO_SETTINGS
 export BASEROW_WEB_FRONTEND_STARTUP_COMMAND="${BASEROW_WEB_FRONTEND_STARTUP_COMMAND:-$DEFAULT_WEB_FRONTEND_STARTUP_COMMAND}"
 export BASEROW_BACKEND_STARTUP_COMMAND="${BASEROW_BACKEND_STARTUP_COMMAND:-$DEFAULT_BACKEND_STARTUP_COMMAND}"
 export BASEROW_CELERY_WORKER_STARTUP_COMMAND="${BASEROW_CELERY_WORKER_STARTUP_COMMAND:-$DEFAULT_CELERY_WORKER_STARTUP_COMMAND}"
+export BASEROW_CELERY_CLT_WORKER_STARTUP_COMMAND="celery-cloudlanguagetoolsworker"
 export BASEROW_CELERY_EXPORT_WORKER_STARTUP_COMMAND="${BASEROW_CELERY_EXPORT_WORKER_STARTUP_COMMAND:-$DEFAULT_CELERY_EXPORT_WORKER_STARTUP_COMMAND}"
 export BASEROW_CELERY_BEAT_STARTUP_COMMAND="${BASEROW_CELERY_BEAT_STARTUP_COMMAND:-$DEFAULT_CELERY_BEAT_STARTUP_COMMAND}"
 export XDG_CONFIG_HOME=/home/$DOCKER_USER/
diff --git a/supervisor/supervisor.conf b/supervisor/supervisor.conf
index 0856cd2fc..a2d9f7918 100644
--- a/supervisor/supervisor.conf
+++ b/supervisor/supervisor.conf
@@ -60,6 +60,19 @@ priority=2
 startretries=3
 startsecs=30
 
+[program:celerycltworker]
+user=%(ENV_DOCKER_USER)s
+directory=/baserow/backend
+command=/baserow/supervisor/wrapper.sh CYAN CLT_WORKER ./docker/docker-entrypoint.sh %(ENV_BASEROW_CELERY_CLT_WORKER_STARTUP_COMMAND)s
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stdout
+stderr_logfile_maxbytes=0
+autostart=true
+priority=2
+startretries=3
+startsecs=30
+
 [program:exportworker]
 user=%(ENV_DOCKER_USER)s
 directory=/baserow/backend
diff --git a/web-frontend/config/nuxt.config.base.js b/web-frontend/config/nuxt.config.base.js
index cc9bb704a..df34aa407 100644
--- a/web-frontend/config/nuxt.config.base.js
+++ b/web-frontend/config/nuxt.config.base.js
@@ -41,6 +41,7 @@ export default function (
       '@nuxtjs/composition-api/module',
     ],
     sentry: {
+      dsn: 'https://33f709910b214ed282315bd91344bae0@o968582.ingest.sentry.io/6742673',
       clientIntegrations: {
         Dedupe: {},
         ExtraErrorData: {},
diff --git a/web-frontend/modules/core/assets/scss/components/auth.scss b/web-frontend/modules/core/assets/scss/components/auth.scss
index 59abdbc91..71efdc4a2 100644
--- a/web-frontend/modules/core/assets/scss/components/auth.scss
+++ b/web-frontend/modules/core/assets/scss/components/auth.scss
@@ -20,7 +20,7 @@
   @extend %mb-24;
 
   & img {
-    height: 100%;
+    width: 100%;
   }
 }
 
diff --git a/web-frontend/modules/core/components/ExternalLinkBaserowLogo.vue b/web-frontend/modules/core/components/ExternalLinkBaserowLogo.vue
index 5060281fa..fba3d37e8 100644
--- a/web-frontend/modules/core/components/ExternalLinkBaserowLogo.vue
+++ b/web-frontend/modules/core/components/ExternalLinkBaserowLogo.vue
@@ -1,12 +1,13 @@
 <template>
   <a
-    href="https://baserow.io"
+    href="https://words.vocab.ai"
     target="_blank"
-    title="Baserow - open source no-code database tool and Airtable alternative"
+    title="Vocab Words - the Language Learning Spreadsheet"
   >
-    <Logo
-      height="14"
-      alt="Baserow - open source no-code database tool and Airtable alternative"
+    <img
+      src="@baserow/modules/core/static/img/vocabai_logo.png"
+      alt="Vocab Words - the Language Learning Spreadsheet"
+      style="width: 110px;"
     />
   </a>
 </template>
diff --git a/web-frontend/modules/core/components/auth/Login.vue b/web-frontend/modules/core/components/auth/Login.vue
index 8c0a3d997..321a684a3 100644
--- a/web-frontend/modules/core/components/auth/Login.vue
+++ b/web-frontend/modules/core/components/auth/Login.vue
@@ -6,7 +6,7 @@
       <div v-if="displayHeader">
         <div class="auth__logo">
           <nuxt-link :to="{ name: 'index' }">
-            <Logo />
+            <img src="@baserow/modules/core/static/img/vocabai_banner.png" alt="" />
           </nuxt-link>
         </div>
         <h1 class="auth__head-title">{{ $t('login.title') }}</h1>
diff --git a/web-frontend/modules/core/components/dashboard/DashboardHelp.vue b/web-frontend/modules/core/components/dashboard/DashboardHelp.vue
index c360b5add..358d49a4e 100644
--- a/web-frontend/modules/core/components/dashboard/DashboardHelp.vue
+++ b/web-frontend/modules/core/components/dashboard/DashboardHelp.vue
@@ -1,78 +1,14 @@
 <template>
-  <Alert
-    v-if="displayAlert"
-    type="blank"
-    close-button
-    position="bottom"
-    :width="396"
-    @close="handleAlertClose"
-  >
-    <template #image
-      ><img
-        src="@baserow/modules/core/assets/images/dashboard_alert_image.png"
-        srcset="
-          @baserow/modules/core/assets/images/dashboard_alert_image@2x.png 2x
-        "
-      />
-    </template>
-    <template #title
-      ><h4>{{ $t('dashboard.alertTitle') }}</h4></template
-    >
-    <p>{{ $t('dashboard.alertText') }}</p>
-    <template #actions>
-      <Button
-        tag="a"
-        href="https://gitlab.com/baserow/baserow"
-        target="_blank"
-        rel="noopener noreferrer"
-        type="secondary"
-        icon="baserow-icon-gitlab"
-      >
-        {{ $t('dashboard.starOnGitlab') }}</Button
-      >
-      <ButtonIcon
-        v-tooltip="$t('dashboard.shareOnTwitter')"
-        tag="a"
-        tooltip-position="top"
-        :href="`https://twitter.com/intent/tweet?url=https://baserow.io&hashtags=opensource,nocode,database,baserow&text=${encodeURI(
-          $t('dashboard.tweetContent')
-        )}`"
-        target="_blank"
-        rel="noopener noreferrer"
-        icon="baserow-icon-twitter"
-      ></ButtonIcon>
-      <ButtonIcon
-        v-tooltip="$t('dashboard.shareOnReddit')"
-        tag="a"
-        tooltip-position="top"
-        icon="baserow-icon-reddit"
-        :href="
-          'https://www.reddit.com/submit?url=https://baserow.io&title=' +
-          encodeURI($t('dashboard.redditTitle'))
-        "
-        target="_blank"
-        rel="noopener noreferrer"
-      ></ButtonIcon>
-      <ButtonIcon
-        v-tooltip="$t('dashboard.shareOnFacebook')"
-        tag="a"
-        tooltip-position="top"
-        icon="baserow-icon-facebook"
-        href="https://www.facebook.com/sharer/sharer.php?u=https://baserow.io"
-        target="_blank"
-        rel="noopener noreferrer"
-      ></ButtonIcon>
-      <ButtonIcon
-        v-tooltip="$t('dashboard.shareOnLinkedIn')"
-        tag="a"
-        tooltip-position="top"
-        icon="baserow-icon-linkedin"
-        href="https://www.linkedin.com/sharing/share-offsite/?url=https://baserow.io"
-        target="_blank"
-        rel="noopener noreferrer"
-      ></ButtonIcon>
-    </template>
-  </Alert>
+  <div class="dashboard__help">
+    <img src="@baserow/modules/core/static/img/vocabai_banner.png" alt="" style="height: 80px;"/>
+    <p class="dashboard__help-description">
+      Introducing <b>Vocab.Ai</b> - The revolutionary language learning spreadsheet designed to supercharge your vocabulary and sentence mastery!
+      <br/>
+      Need help ? <a href="https://lucw.medium.com/unlock-the-power-of-pinyin-vocab-ais-edc5db83a1a9" target="_blank">Chinese and Pinyin tutorial</a>
+      <br/>
+      <i>note: Vocab.Ai is based on open-source database <a href="https://baserow.io" target="_blank">Baserow</a></i>
+    </p>
+  </div>
 </template>
 
 <script>
diff --git a/web-frontend/modules/core/head.js b/web-frontend/modules/core/head.js
index a20ca13fa..5ba1f791b 100644
--- a/web-frontend/modules/core/head.js
+++ b/web-frontend/modules/core/head.js
@@ -1,6 +1,6 @@
 export default {
-  title: 'Baserow',
-  titleTemplate: '%s | Baserow',
+  title: 'Vocab Words',
+  titleTemplate: '%s // Vocab Words',
   meta: [
     { charset: 'utf-8' },
     {
diff --git a/web-frontend/modules/core/plugins/posthog.js b/web-frontend/modules/core/plugins/posthog.js
index e87e94aa8..2820da435 100644
--- a/web-frontend/modules/core/plugins/posthog.js
+++ b/web-frontend/modules/core/plugins/posthog.js
@@ -31,12 +31,8 @@ export default function ({ app: { router, $config, store } }, inject) {
       // Check if the user identification on every page route because they could
       // have changed accounts. This keeps the Posthog code isolated from the
       // authenticated system.
-      if (
-        isAuthenticated &&
-        userId &&
-        userId.toString() !== posthog.get_distinct_id()
-      ) {
-        posthog.identify(userId, { user_email: userEmail })
+      if (isAuthenticated && userId !== posthog.get_distinct_id()) {
+        posthog.identify(userEmail, { user_email: userEmail })
       }
 
       // Some pages must not be tracked like the ones that can expose a sensitive token.
